src/gluonts/torch/model/mqf2/module.py

```diff
@@ -0,0 +1,259 @@
+            lags_seq=lags_seq,
+            scaling=scaling,
+            num_parallel_samples=num_parallel_samples,
+        )
+
+        self.threshold_input = threshold_input
+        self.es_num_samples = es_num_samples
+
+        convexnet = PICNN(
+            dim=prediction_length,
+            dimh=icnn_hidden_size,
+            dimc=hidden_size,
+            num_hidden_layers=icnn_num_layers,
+            symm_act_first=True,
+        )
+        deepconvexnet = DeepConvexNet(
+            convexnet,
+            prediction_length,
+            is_energy_score=is_energy_score,
+            estimate_logdet=estimate_logdet,
+        )
+
+        if is_energy_score:
+            networks = [deepconvexnet]
+        else:
+            networks = [
+                ActNorm(prediction_length),
+                deepconvexnet,
+                ActNorm(prediction_length),
+            ]
+
+        self.picnn = SequentialNet(networks)
+
+    def unroll_lagged_rnn(
+        self,
+        feat_static_cat: torch.Tensor,
+        feat_static_real: torch.Tensor,
+        past_time_feat: torch.Tensor,
+        past_target: torch.Tensor,
+        past_observed_values: torch.Tensor,
+        future_time_feat: Optional[torch.Tensor] = None,
+        future_target: Optional[torch.Tensor] = None,
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        """
+        Unrolls the RNN encoder over the context window of the time series
+        Returns the hidden state of the RNN and the scale
+
+        Parameters
+        ----------
+        feat_static_cat
+            static categorial features (batch_size, num_feat_static_cat)
+        feat_static_real
+            static real-valued features (batch_size, num_feat_static_real)
+        past_time_feat
+            Past time features (batch_size, history_length, num_features)
+        past_target
+            Past target values (batch_size, history_length)
+        past_observed_values
+            Indicate whether or not the values were observed
+            (batch_size, history_length)
+        future_time_feat
+            Future time features (batch_size, prediction_length, num_features)
+        future_target
+            Future target values (batch_size, prediction_length)
+
+        Returns
+        -------
+        hidden_state
+            RNN hidden state (batch_size, context_length, hidden_size)
+        scale
+            Scale calculated from the context window (batch_size, 1)
+        """
+
+        _, scale, hidden_state, _, _ = super().unroll_lagged_rnn(
+            feat_static_cat=feat_static_cat,
+            feat_static_real=feat_static_real,
+            past_time_feat=past_time_feat,
+            past_target=past_target,
+            past_observed_values=past_observed_values,
+            future_time_feat=future_time_feat,
+            future_target=future_target,
+        )
+
+        hidden_state = hidden_state[:, : self.context_length]
+
+        return hidden_state, scale
```

Comment:
It appears like this method is doing very little on top of the one defined in the `super` class (basically just slicing the returned `hidden_state`). It's also changing the signature quite a bit wrt to the `super` method, by reducing the amount of stuff returned.

In my opinion, it would be better not to define this method at all, and instead invoke directly the one from `super` followed by slicing.

Reply:
Yes I should return the all the outputs as in the superclass. I will make changes about that. 

Basically this function unrolls the rnn but not all the way until the end of the time series. So I do the slicing to chop off the things I don't need here. I think that it would be easier for the users to not slice the hidden state every time they call the function.

Reply:
Isn't it the case that by just not providing the "future" data (`future_time_feat` and `future_target`) you already achieve this? I.e., you only get out the first `context_length` hidden state as output?

I'm not 100% sure, but you may want to check that, to save even more code